File: countNodesBfs.py
from collections import deque

# Definition for TreeNode
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def countNodes(self, root: TreeNode) -> int:
        # Empty tree
        if not root:
            return 0

        # Queue
        qu = deque([root])

        ans = 0

        while qu:
            sz = len(qu)

            for i in range(sz):
                if qu[0].left:
                    qu.append(qu[0].left)
                if qu[0].right:
                    qu.append(qu[0].right)

                qu.popleft()
                ans += 1

        return ans

File: countNodesBfs.cpp
#include <bits/stdc++.h>
using namespace std;


//Definition for treeNode
struct TreeNode{
   int val;
   TreeNode*left;
   TreeNode*right;
   TreeNode(int x): val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
     int countNodes(TreeNode* root){
         //Empty tree
         if(root == NULL)return 0;

         //queue
         queue<TreeNode*>qu;

         qu.push(root);

         int ans = 0;

         while(!qu.empty()){

             int sz= qu.size();

             for(int i=1;i<=sz;i++){
                 if(qu.front()->left)qu.push(qu.front()->left);
                 if(qu.front()->right)qu.push(qu.front()->right);

                 qu.pop();
                 ans++;
             }
         }

         return ans;
     }
};
